fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRHttpServletRequestWrapper.java

```diff
@@ -315,6 +341,19 @@ private String possiblyMapQueryParameterValue(String headerName, String value) {
                 else if(delegateQueryString.contains("_format=application/fhir+xml")) {
                     value = FHIRMediaType.APPLICATION_FHIR_XML;
                 }
+                
+                // if there is charset defined in _format, then take it.
+                if (delegateQueryString.contains("_format=application/fhir+json") 
```

Comment:
doubling up on checks on the contains method.  
embed in existing else/if blocks. 
use indexOf on the delegateQueryString to get the initial index, and check to see if the following char is ';' or non-existant

something like this... 
int loc = delegateQueryString.indexOf("_format="); 
then 
if (loc > 0){
  // change between the four cases. 
  
}

Reply:
per discussion the _format decode/false in this case also works

Reply:
I don't use indexOf, because we have to consider if charset is the first or the last format or in the middle of the _format string

Reply:
changed to use existing contains if else blocks.